grpc-native/src/main/java/org/ballerinalang/net/grpc/builder/BallerinaFileBuilder.java

```diff
@@ -198,61 +180,52 @@ private void computeSourceContent(byte[] descriptor, String mode) throws CodeBui
 
                     if (method.containsEmptyType() && !(stubFileObject.isMessageExists(EMPTY_DATA_TYPE))
                             && !hasEmptyMessage) {
-                        Message message = Message.newBuilder(EmptyMessage.newBuilder().getDescriptor().toProto())
-                                .build();
+                        Message message =
+                                Message.newBuilder(getFileDescriptor(EMPTY_PROTO_PACKAGE_KEY).getMessageTypes().get(0)
+                                        .toProto()).build();
                         messageList.add(message);
                         stubFileObject.addMessage(message);
                         hasEmptyMessage = true;
                     }
                 }
                 stubFileObject.addServiceStub(serviceStubBuilder.build());
-                stubFileObject.setEnumList(enumList);
-                stubFileObject.setDescriptors(descriptors);
-                if (!stubRootDescriptor.isEmpty()) {
-                    stubFileObject.setRootDescriptor(stubRootDescriptor);
-                }
+            }
+            // update enum message types in stub file object
+            stubFileObject.setEnumList(enumList);
+            // update descriptors in stub file object
+            stubFileObject.setDescriptors(descriptors);
+            if (!stubRootDescriptor.isEmpty()) {
+                stubFileObject.setRootDescriptor(stubRootDescriptor);
+            }
 
+            int serviceIndex = 0;
+            for (ServiceStub serviceStub : stubFileObject.getStubList()) {
                 if (GRPC_SERVICE.equals(mode)) {
-                    serviceFile = sampleServiceBuilder.build();
-                    serviceFile.setEnableEp(enableEp);
-                    if (enableEp) {
-                        enableEp = false;
-                    }
-
-                    String servicePath = generateOutputFile(this.balOutPath, serviceDescriptor.getName() +
+                    String serviceFilePath = generateOutputFile(this.balOutPath, serviceStub.getServiceName() +
                             SAMPLE_SERVICE_FILE_PREFIX);
-                    writeOutputFile(serviceFile, SAMPLE_SERVICE_TEMPLATE_NAME, servicePath);
-
-                    String stubFilePath = generateOutputFile(this.balOutPath, filename + STUB_FILE_PREFIX);
-                    writeOutputFile(stubFileObject, SERVICE_STUB_TEMPLATE_FILE_NAME, stubFilePath);
-                } else if (GRPC_CLIENT.equals(mode)) {
-                    String clientFilePath = generateOutputFile(
-                            this.balOutPath,
-                            serviceDescriptor.getName() + SAMPLE_FILE_PREFIX
+                    writeOutputFile(SyntaxTreeGenerator.generateSyntaxTreeForServiceSample(serviceStub,
+                            serviceIndex == 0), serviceFilePath);
+                }
+                if (GRPC_CLIENT.equals(mode)) {
```

Comment:
Can't we use the `else if` block here since the mode cannot have client and service at the same time?

Reply:
Fixed in e7ee2bc